java/src/lib.rs

```diff
@@ -122,8 +97,21 @@ pub extern "system" fn Java_glide_ffi_resolvers_RedisValueResolver_valueFromPoin
     _class: JClass<'local>,
     pointer: jlong,
 ) -> JObject<'local> {
-    let value = unsafe { Box::from_raw(pointer as *mut Value) };
-    redis_value_to_java(&mut env, *value, true)
+    handle_panics(
+        move || {
+            fn value_from_pointer<'a>(
+                env: &mut JNIEnv<'a>,
+                pointer: jlong,
+            ) -> Result<JObject<'a>, FFIError> {
+                let value = unsafe { Box::from_raw(pointer as *mut Value) };
+                redis_value_to_java(env, *value, true)
+            }
+            let result = value_from_pointer(&mut env, pointer);
```

Comment:
What is the benefit of defining a funcion and calling it right away?
Why can't we do this?
```suggestion
            let value = unsafe { Box::from_raw(pointer as *mut Value) };
            let result = redis_value_to_java(env, *value, true);
```

Reply:
In this case, there is no particular benefit. The reason why I define inner functions is because I didn't want to have to handle errors inline every single time I get a Result back (instead of using `unwrap`). To use `?` syntax, you need a function that returns a `Result` though, so this is sort of a framework to follow. As for why I couldn't just roll this into another closure that is called from the closure passed to `catch_unwind`, it's because of this issue: https://github.com/jni-rs/jni-rs/issues/432

Basically, the author of jni-rs didn't implement UnwindSafe for &mut JNIEnv, which means in order to use the JNIEnv, we need to own it inside of the closure instead of just borrowing it. If I were to pass the inner function as a closure, it would need to either borrow the JNIEnv (we can't do that), or always explicitly return the JNIEnv along with every payload, which is just unergonomic. As far as I can tell, there isn't a cleaner solution than this as long as that issue is still open. I could move the inner function definitions to the top level instead to avoid rightward drift, but that's just a style change.

Reply:
Is it difficutl to
> implement UnwindSafe for &mut JNIEnv

?
Maybe ~we~ you can contribute to `jni-rs` and later update our `lib.rs` once this fix released.

Reply:
It's not possible to do this in our code because foreign traits cannot be implemented for foreign types. It would break trait coherence rules. It would need to be fixed upstream in `jni-rs`. I don't know how difficult it would be to implement this upstream. I'll need to try and see what goes wrong, but that's not a bad idea.

Reply:
Yes I considered pushing a fix to `jni-rs`, but we shoudn't wait for it now.